src/lib/utils/hooks/SynapseAPI/dataaccess/useDataAccessSubmission.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  useInfiniteQuery,
+  UseInfiniteQueryOptions,
+  useMutation,
+  UseMutationOptions,
+  useQuery,
+  useQueryClient,
+  UseQueryOptions,
+} from 'react-query'
+import { SynapseClient } from '../../..'
+import { SynapseClientError } from '../../../SynapseClient'
+import { useSynapseContext } from '../../../SynapseContext'
+import { Submission } from '../../../synapseTypes/AccessRequirement/Submission'
+import { SubmissionStateChangeRequest } from '../../../synapseTypes/AccessRequirement/SubmissionStateChangeRequest'
+import {
+  SubmissionSearchRequest,
+  SubmissionSearchResponse,
+} from '../../../synapseTypes/AccessSubmission'
+
+const dataAccessSubmissionQueryKeys = {
+  submission: (id: string | number) => ['dataAccessSubmission', id],
+  search: (params?: SubmissionSearchRequest) =>
+    params ? ['accessSubmissionSearch', params] : ['accessSubmissionSearch'],
+}
+
+export default function useGetDataAccessSubmission(
+  submissionId: string | number,
+  options?: UseQueryOptions<Submission, SynapseClientError>,
+) {
+  const { accessToken } = useSynapseContext()
+
+  return useQuery<Submission, SynapseClientError>(
+    dataAccessSubmissionQueryKeys.submission(submissionId),
+    () => SynapseClient.getSubmissionById(submissionId, accessToken),
+    options,
+  )
+}
+
+export function useSearchAccessSubmissionsInfinite(
+  params: SubmissionSearchRequest,
+  options?: UseInfiniteQueryOptions<
+    SubmissionSearchResponse,
+    SynapseClientError
+  >,
+) {
+  const { accessToken } = useSynapseContext()
+
+  return useInfiniteQuery<SubmissionSearchResponse, SynapseClientError>(
+    dataAccessSubmissionQueryKeys.search(params),
+    async context => {
+      return await SynapseClient.searchAccessSubmission(
+        {
+          ...params,
+          nextPageToken: context.pageParam,
+        },
+        accessToken,
+      )
+    },
+    {
+      ...options,
+      getNextPageParam: page => page.nextPageToken,
+    },
+  )
+}
+
+export function useUpdateDataAccessSubmissionState(
+  options?: UseMutationOptions<
+    Submission,
+    SynapseClientError,
+    SubmissionStateChangeRequest
+  >,
+) {
+  const queryClient = useQueryClient()
+  const { accessToken } = useSynapseContext()
+
+  return useMutation<
+    Submission,
+    SynapseClientError,
+    SubmissionStateChangeRequest
+  >(
+    (request: SubmissionStateChangeRequest): Promise<Submission> =>
+      SynapseClient.updateSubmissionStatus(request, accessToken),
+    {
+      ...options,
+      onSuccess: async (updatedSubmission, variables, ctx) => {
+        // Invalidate all searches, since updating the status will affect filtered search results
+        queryClient.invalidateQueries(dataAccessSubmissionQueryKeys.search())
```

Comment:
Including the search that determines the number of open requests shown in the NavDrawer